metagraph/src/cli/query.hpp

```diff
@@ -28,24 +29,43 @@ class Config;
 
 using StringGenerator = std::function<void(std::function<void(const std::string &)>)>;
 
+/**
+ * Construct a query graph
+ * @param anno_graph the input annotated de Bruijn graph
+ * @param call_sequences generate sequences to be queried against anno_graph
+ * @param discovery_fraction the minimum fraction at which a generated sequence
+ *                           must be present in anno_graph to be considered
+ * @param num_threads number of threads to use
+ * @param canonical if true, the returned query graph is a canonical graph
+ * @param sub_k also include k-mers whose suffixes of length sub_k match sub_k-mers
+ *              in the generated sequences. This is only applicable for DBGSuccinct
```

Comment:
There's a command line argument `--align-max-num-seeds-per-locus` which is set to 1 by default.

For a given k-mer, it finds the max k' s.t. there exist n > 0 nodes with a suffix match of length k'. If n > max-num-seeds-per-locus, then it ignores that k-mer and moves on (as a way to prevent too many seeds from being processed).